From 4ed55bdc79542e795243043f7783026829a9f8b8 Mon Sep 17 00:00:00 2001
From: Lyude Paul <lyude@redhat.com>
Date: Mon, 18 Nov 2024 15:59:34 -0500
Subject: [PATCH 020/109] rust: helpers/drm.c: Split into its own folder

There's not that many bindings in here right now, but we're about to add
quite a lot more - which means it's probably worth it to further split out
the current helpers we have in helpers/drm.c into their own folder.

Signed-off-by: Lyude Paul <lyude@redhat.com>
(cherry picked from commit 9c1a257a0278428693f0f1a2b9a1804c141b62f5)
---
 rust/helpers/drm/drm.c            |  6 ++++++
 rust/helpers/{drm.c => drm/gem.c} | 10 ----------
 rust/helpers/drm/vma_manager.c    |  9 +++++++++
 rust/helpers/helpers.c            |  2 +-
 4 files changed, 16 insertions(+), 11 deletions(-)
 create mode 100644 rust/helpers/drm/drm.c
 rename rust/helpers/{drm.c => drm/gem.c} (88%)
 create mode 100644 rust/helpers/drm/vma_manager.c

diff --git a/rust/helpers/drm/drm.c b/rust/helpers/drm/drm.c
new file mode 100644
index 000000000000..05ecbf1a0161
--- /dev/null
+++ b/rust/helpers/drm/drm.c
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#ifdef CONFIG_DRM
+#include "gem.c"
+#include "vma_manager.c"
+#endif
diff --git a/rust/helpers/drm.c b/rust/helpers/drm/gem.c
similarity index 88%
rename from rust/helpers/drm.c
rename to rust/helpers/drm/gem.c
index 65f3f22b0e1d..78ce4bda5b4f 100644
--- a/rust/helpers/drm.c
+++ b/rust/helpers/drm/gem.c
@@ -2,9 +2,6 @@
 
 #include <drm/drm_gem.h>
 #include <drm/drm_gem_shmem_helper.h>
-#include <drm/drm_vma_manager.h>
-
-#ifdef CONFIG_DRM
 
 __rust_helper void rust_helper_drm_gem_object_get(struct drm_gem_object *obj)
 {
@@ -16,12 +13,6 @@ __rust_helper void rust_helper_drm_gem_object_put(struct drm_gem_object *obj)
 	drm_gem_object_put(obj);
 }
 
-__rust_helper __u64
-rust_helper_drm_vma_node_offset_addr(struct drm_vma_offset_node *node)
-{
-	return drm_vma_node_offset_addr(node);
-}
-
 #ifdef CONFIG_DRM_GEM_SHMEM_HELPER
 __rust_helper void
 rust_helper_drm_gem_shmem_object_free(struct drm_gem_object *obj)
@@ -75,4 +66,3 @@ rust_helper_drm_gem_shmem_object_mmap(struct drm_gem_object *obj, struct vm_area
 }
 
 #endif /* CONFIG_DRM_GEM_SHMEM_HELPER */
-#endif /* CONFIG_DRM */
diff --git a/rust/helpers/drm/vma_manager.c b/rust/helpers/drm/vma_manager.c
new file mode 100644
index 000000000000..2bea0aa69f5c
--- /dev/null
+++ b/rust/helpers/drm/vma_manager.c
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <drm/drm_vma_manager.h>
+
+__rust_helper __u64
+rust_helper_drm_vma_node_offset_addr(struct drm_vma_offset_node *node)
+{
+	return drm_vma_node_offset_addr(node);
+}
diff --git a/rust/helpers/helpers.c b/rust/helpers/helpers.c
index 0d85b5e68ec2..41fca1ea87c2 100644
--- a/rust/helpers/helpers.c
+++ b/rust/helpers/helpers.c
@@ -59,7 +59,7 @@
 #include "device.c"
 #include "dma.c"
 #include "dma-resv.c"
-#include "drm.c"
+#include "drm/drm.c"
 #include "drm_gpuvm.c"
 #include "err.c"
 #include "irq.c"
